includes/class-wc-payments-upe-blocks-payment-method.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * Class WC_Payments_Blocks_Payment_Method
+ *
+ * @package WooCommerce\Payments
+ */
+
+use Automattic\WooCommerce\Blocks\Payments\Integrations\AbstractPaymentMethodType;
+use WCPay\Payment_Methods\UPE_Payment_Gateway;
+
+/**
+ * The payment method, which allows the gateway to work with WooCommerce Blocks.
+ */
+class WC_Payments_UPE_Blocks_Payment_Method extends AbstractPaymentMethodType {
+	/**
+	 * Initializes the class.
+	 */
+	public function initialize() {
+		$this->name    = UPE_Payment_Gateway::GATEWAY_ID;
+		$this->gateway = WC_Payments::get_gateway();
+	}
+
+	/**
+	 * Checks whether the gateway is active.
+	 *
+	 * @return boolean True when active.
+	 */
+	public function is_active() {
+		return $this->gateway->is_available();
+	}
+
+	/**
+	 * Defines all scripts, necessary for the payment method.
+	 *
+	 * @return string[] A list of script handles.
+	 */
+	public function get_payment_method_script_handles() {
+		wp_register_script(
+			'stripe',
+			'https://js.stripe.com/v3/',
+			[],
+			'3.0',
+			true
+		);
+
+		wp_register_script(
+			'WCPAY_BLOCKS_CHECKOUT',
+			plugins_url( 'dist/upe-blocks-checkout.js', WCPAY_PLUGIN_FILE ),
+			[ 'stripe' ],
+			'1.0.1',
+			true
+		);
```

Comment:
I guess I don't really mind the presence of this class for now, but could you just have inherited from the `WC_Payments_Blocks_Payment_Method` class, since it appears this is the only part that differs. `UPE_Payment_Gateway::GATEWAY_ID` and `WC_Payment_Gateway_WCPay::GATEWAY_ID` both return the same id string, so that seems a little redundant inside the `initialize()` function. But please correct me if I'm being short-sighted and missing anything here!

Reply:
Good call, I was really torn on this one, so with your nudge, I went the route of inheritance now with the latest push.